File: Classwork/AbstractClass/Square.java
/**
 *  filename: Square.java
 *  purpose: extend the AbstractShape class
 *  author: Dr. Johnson
 *  date: 2020-08-18
 */

public class Square extends Rectangle {

   public Square( int s ) {
      super( s, s );
      System.out.println( "\n  The extended class 'Square' constructor is called." );
   }

}

File: Classwork/classwork15/Square.py
###
# filename: Square.py
# purpose: demonstration of subclassing
# author:   Dr. Johnson
# date:     2024-04-16
###

import math

# Define a Square subclass from Polygon
#  Overrides the 'area()' from the superclass
class Square( Polygon ):
   def __init__( self, side ):
      super().__init__( 4 )
      self.sideLength = side

   def area( self ):
      return self.sideLength ** 2

